hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/HttpClientsFactory.java

```diff
@@ -43,7 +42,9 @@ public HttpClient createClientForHttp1(String name) {
                 configFactory.getIntProperty(CONSUMER_HTTP_CLIENT_THREAD_POOL_SIZE),
                 configFactory.getBooleanProperty(CONSUMER_HTTP_CLIENT_THREAD_POOL_MONITORING));
 
-        HttpClient client = new HttpClient(createSslContextFactory());
+        HttpClient client = sslContextFactoryProvider.provideSslContextFactory()
+                .map(sslContextFactory -> new HttpClient(sslContextFactory))
+                .orElseGet(() -> new HttpClient());
```

Comment:
It could be simplified to:
```
.map(HttpClient::new)
.orElseGet(HttpClient::new);
```
but I think it would be less understandable.
What is your opinion?

Reply:
I prefer the committed version. :)